python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -273,7 +273,8 @@ def _restart_worker_group(
         Returns:
             True if the worker group was successfully restarted, False otherwise.
         """
-        self._worker_group.shutdown()
+        if self._worker_group:
+            self._shutdown_worker_group()
```

Comment:
After the controller refactor, I think we can remove this shutdown and make this method `_start_worker_group` instead, and assert not worker_group here. Since the `_restart_worker_group` is only called in one place now (the execute resize decision logic). Feel free to leave as a follow-up.

Reply:
Oh that's neat.  Let me see if it's a quick fix to do in this PR.